content/en/docs/architecture/ci-operator-internals/testing.md

```diff
@@ -0,0 +1,425 @@
+configuration is done using `base_images`:
+
+{{< highlight yaml >}}
+base_images:
+  fedora:
+    namespace: ci
+    name: fedora
+    tag: latest
+tests:
+- as: test
+  commands: echo test
+  container:
+    from: fedora
+    clone: false
+resources:
+  # as before
+{{< / highlight >}}
+
+`clone: false` was added since by default tests which use one of the
+`base_images` have the source code of the repository automatically cloned.  Our
+configuration file is finally valid and ready to guide the execution of the test
+it defines, but a few topics need to be covered before we can see the result.
+
+## Test namespaces
+
+Even though `ci-operator` is being executed directly on our machine, that is not
+where most of the actual work is done.  Image builds, test executions, etc. must
+happen inside an Openshift cluster, and `ci-operator` expects to be executed in
+an environment where it has access to one.  This is why this final error message
+mentions "in-cluster configuration" and missing Kuberentes host/port
+information.  As mentioned before, the default environment is assumed to be a
+Prow job, which is executed inside a Kubernetes `Pod`.  There, those environment
+variables are provided automatically by Kubernetes (see the [service
+documentation][services_networking]).
+
+Outside of a Kubernetes cluster, the connection information can be passed using
+the standard `KUBECONFIG` environment variable, which should point to a file
+just as `kubectl` and/or `oc` would expect it.  Note that the default context
+will be used, so make sure it points to the correct cluster.
+
+{{< highlight bash >}}
+$ oc whoami
+error: Missing or incomplete configuration info.  Please point to an existing, complete config file:
+
+
+  1. Via the command-line flag --kubeconfig
+  2. Via the KUBECONFIG environment variable
+  3. In your home directory as ~/.kube/config
+
+To view or setup config directly use the 'config' command.
+$ export KUBECONFIG=…
+$ oc whoami --show-context
+bbguimaraes/api-build01-ci-devcluster-openshift-com:6443/bbarcaro
+{{< / highlight >}}
+
+Our `ci-operator` is finally ready to execute its test, but one last adjustment
+will make it a more pleasant experience.  In Kubernetes, isolation between
+workloads is achieved using [_namespaces_][namespaces].  Openshift wraps
+upstream namespaces into its own resource, the [_project_][project].  For most
+purposes, the distinction is irrelevant and both names are used interchangeably.
+`ci-operator` uses an ephemeral namespace to guarantee no interference between
+unrelated tests, which is referred to as the _test namespace_.  The process by
+which this namespace is set up will be discussed later; of relevance here is
+that this setup is inconvenient for manual testing, for various reasons.
+
+We can create our own test namespace in advance and simply ask `ci-operator` to
+use it instead of attempting to create one.  The build clusters are shared
```

Comment:
That's what I use (mostly because I haven't changed my `kubeconfig` from the days we only had `build01` =), but any cluster will do.

This is meant for Test Platform, so if someone doesn't have permission, they should be given.

Reply:
> This is meant for Test Platform

Gocha